osu.Game/Screens/Edit/Compose/Components/BlueprintContainer.cs

```diff
@@ -115,19 +115,21 @@ private void load()
 
         protected override bool OnMouseDown(MouseDownEvent e)
         {
-            bool selectionPerformed = performMouseDownActions(e);
+            var selectionBefore = SelectionHandler.SelectedItems.ToArray();
+            bool handled = performMouseDownActions(e);
+            bool selectionChanged = !SelectionHandler.SelectedItems.SequenceEqual(selectionBefore);
             bool movementPossible = prepareSelectionMovement(e);
 
-            // check if selection has occurred
-            if (selectionPerformed)
+            if (selectionChanged)
             {
-                // only unmodified right click should show context menu
+                // if the selection changed and there are no modifiers pressed, don't block so the context menu still shows.
                 bool shouldShowContextMenu = e.Button == MouseButton.Right && !e.ShiftPressed && !e.AltPressed && !e.SuperPressed;
-
-                // stop propagation if not showing context menu
                 return !shouldShowContextMenu;
             }
 
+            if (handled)
+                return true;
```

Comment:
As per test failures, this prevents the context menu from showing at all on selected blueprints :grimacing:

Reply:
Pretty obvious why.. luckily the fix is using the same conditions I used elsewhere in this change.